staging/src/k8s.io/kubelet/config/v1beta1/types.go

```diff
@@ -824,6 +838,14 @@ type KubeletConfiguration struct {
 	// Default: "10s"
 	// +optional
 	ShutdownGracePeriodCriticalPods metav1.Duration `json:"shutdownGracePeriodCriticalPods,omitempty"`
+	// Reserved Memory specifies a comma separated list of memory reservations for NUMA nodes.
+	// Each configuration describes the reserved memory for desired memory types on a specific NUMA node.
+	// The Memory Manager validates whether total amount of reserved memory is identical to memory reserved by the Node Allocatable feature.
+	// The format is [{"numaNode": integer, "limits": {"type": "limit"}}, {"numaNode": integer, "limits": {"type": "limit"}}, ...]
```

Comment:
The format line no longer makes sense since this is a struct. You should describe the validation rules for clients - for instance, can a user specify duplicate numa nodes?  Can a user provide a memory reservation without specifying limits?  What values are allowed?

Reply:
Can you please point where can I specify validation logic?
- no duplicates NUMA nodes
- no empty limits
- limit can be only the quantatity

Reply:
Regarding the format, I just specified it the same way as I will specify it under the KubeletConfig JSON file, do you have a better representation for the format? I will happily change it :)

Reply:
Generally you would describe it in terms of what a human would do "One entry per numa node that you wish to set limits on."  You don't generally embed the actual JSON here because an example wouldf be better in human terms as:

> For example, to reserve 512Mi only on the first numa node (node 0), you would define a memory reservation for node 0 and a limit on "memory" of "512Mi".

You would also try to answer questions that a human would ask - for instance:

* Must the NUMA nodes be contiguous?
* If I don't specify a NUMA node, what happens?
* What happens if I don't specify any reservations here?
* How does this interact with other nodes?

You wouldn't put the questions themselves in the description, but every question I've asked in this PR is roughly "how would I use this field" and if the answer isn't in the API doc the user has to leave context and go somewhere else.  You don't have to walk through a full example, but any validation rule that is likely to fail a user should be explained in this api doc right here ("Duplicate NUMA nodes are not allowed.  Each entry must have at least one limit set.  NUMA nodes are not required to be contiguous or in order, but any omitted NUMA nodes will not be allowed to be used for ...")

Reply:
Thanks for the clarification, I updated the API doc.